src/Catalog/Helpers/UriUtils.cs

```diff
@@ -83,5 +85,18 @@ public static Uri GetNonhijackableUri(Uri originalUri)
 
             return nonhijackableUri;
         }
+
+        /// <summary>
+        /// Makes a web request to <paramref name="uri"/> and returns <see cref="HttpRequestMessage.RequestUri"/> of the <see cref="HttpResponseMessage.RequestMessage"/>.
+        /// In other words, if <paramref name="uri"/> redirects to another address, this method returns the address that was redirected to.
+        /// </summary>
+        public static async Task<Uri> GetRedirectedRequestMessageUri(Uri uri)
+        {
+            using (var client = new HttpClient())
```

Comment:
Constantly creating `HttpClient` can cause issues. See [this](https://github.com/mspnp/performance-optimization/blob/master/ImproperInstantiation/docs/ImproperInstantiation.md) and [this](https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/). Can you cache this instead?